services/wallet/api.go

```diff
@@ -742,11 +755,179 @@ func (api *API) CreateMultiTransaction(ctx context.Context, multiTransactionComm
 	return nil, api.s.transactionManager.SendTransactionForSigningToKeycard(ctx, cmd, data, api.router.GetPathProcessors())
 }
 
+func updateFields(sd *responses.SendDetails, inputParams requests.RouteInputParams) {
+	sd.SendType = int(inputParams.SendType)
+	sd.FromAddress = types.Address(inputParams.AddrFrom)
+	sd.ToAddress = types.Address(inputParams.AddrTo)
+	sd.FromToken = inputParams.TokenID
+	sd.ToToken = inputParams.ToTokenID
+	sd.FromAmount = inputParams.AmountIn.String()
+	sd.ToAmount = inputParams.AmountOut.String()
+	sd.OwnerTokenBeingSent = inputParams.TokenIDIsOwnerToken
+	sd.Username = inputParams.Username
+	sd.PublicKey = inputParams.PublicKey
+	sd.PackID = inputParams.PackID.String()
+}
+
+func (api *API) BuildTransactionsFromRoute(ctx context.Context, buildInputParams *requests.RouterBuildTransactionsParams) {
+	log.Debug("[WalletAPI::BuildTransactionsFromRoute] builds transactions from the generated best route", "uuid", buildInputParams.Uuid)
+
+	go func() {
+		api.router.StopSuggestedRoutesAsyncCalculation()
+
+		var err error
+		response := &responses.RouterTransactionsForSigning{
+			SendDetails: &responses.SendDetails{
+				Uuid: buildInputParams.Uuid,
+			},
+		}
+
+		defer func() {
+			if err != nil {
+				api.s.transactionManager.ClearLocalRouterTransactionsData()
```

Comment:
I'm not super happy about status-go storing all this "temporary" data (then we need to introduce lifetime management and be super careful about when to clear that so we don't reach some inconsistent state), but as an improvement to make it less error-prone we could make sure all these temporarily cached things are tied to a specific UUID so we can be fully certain whatever we retrieve from the cache belongs to a particular Send/Swap/Bridge operation. No use case for now, but that would even probably let us lift the restriction of "only one route/transaction flow at a time".

Reply:
Well, that's the price of having a code organization like we have.

The sending Tx flow has to be done in a few steps (calculate the route, build tx, sign tx, send tx) there are 2 approaches:
- to return everything from each call and ask for all those params in the next step of the flow
- to keep states on the statusgo side and just let the client call the next step with certain uuid

Since the statusgo code is shared between desktop and mobile app, the first approach may result in what we have now, different client-side handlings for achieving the same result. That approach requires more lines of code than the second approach for sure. In the second approach, all that the client needs to do is to call functions in a certain order with uuid used when it was generating the best route, that's why I think the second approach will force clients to use the flow in the same way and will be easier for future maintaining. 

About sending flow data management on the statusgo side... route's data are cleared in the following cases:
- either on error when generating the best route (handled by statusgo)
- or asking for a new route (handled by statusgo)
- or explicitly by the client calling `StopSuggestedRoutesAsyncCalculation` (calling this func multiple times is fine)
- or when proceeding with sending (handled by statusgo)

clearing transactions' data is fully managed by the statusgo, data are cleared:
- either on error while building the Tx from the route
- or on error while sending the Tx
- or when the Tx is successfully sent

All that minimizes the burden on the client side.